eZ/Publish/Core/Repository/ProxyFactory/ProxyDomainMapper.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Repository\ProxyFactory;
+
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Language;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\API\Repository\Values\Content\Section;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\API\Repository\Values\ContentType\ContentTypeGroup;
+use eZ\Publish\API\Repository\Values\User\User;
+use ProxyManager\Factory\LazyLoadingValueHolderFactory;
+use ProxyManager\Proxy\LazyLoadingInterface;
+
+/**
+ * @internal
+ */
+final class ProxyDomainMapper implements ProxyDomainMapperInterface
+{
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
+
+    /** @var \ProxyManager\Factory\LazyLoadingValueHolderFactory */
+    private $factory;
+
+    public function __construct(Repository $repository, LazyLoadingValueHolderFactory $lazyLoadingValueHolderFactory)
+    {
+        $this->repository = $repository;
+        $this->factory = $lazyLoadingValueHolderFactory;
+    }
+
+    public function createContentProxy(
+        int $contentId,
+        array $prioritizedLanguages = Language::ALL,
+        bool $useAlwaysAvailable = true
+    ): Content {
+        $initializer = function (
+            &$wrappedObject, LazyLoadingInterface $proxy, $method, array $parameters, &$initializer
+        ) use ($contentId, $prioritizedLanguages, $useAlwaysAvailable): bool {
+            $initializer = null;
+            $wrappedObject = $this->repository->getContentService()->loadContent(
```

Comment:
Not a huge fan of that. I prefer pure DI over Repository as the later one is a container inside a Container. Planning to deprecate those methods at some point. It's true of course that this will cause a need to inject a lot of services here, but maybe it's an indication that it requires splitting responsibility? ;)

Getting rid of `$this->repository->get*Service` calls at least would be good if splitting is too much.